poetry/console/commands/plugin/add.py

```diff
@@ -0,0 +1,263 @@
+                constraint = constraint["version"]
+
+            poetry_dependency_section[plugin["name"]] = constraint
+            plugin_names.append(plugin["name"])
+
+        env_dir.joinpath("pyproject.toml").write_text(
+            tomlkit.dumps(pyproject), encoding="utf-8"
+        )
+
+        # From this point forward, all the logic will be deferred to
+        # the update command, by using the previously created `pyproject.toml`
+        # file.
+        application = cast("Application", self.application)
+        update_command: "UpdateCommand" = cast(
+            "UpdateCommand", application.find("update")
+        )
+        # We won't go through the event dispatching done by the application
+        # so we need to configure the command manually
+        update_command.set_poetry(Factory().create_poetry(env_dir))
+        update_command.set_env(system_env)
+        application._configure_installer(update_command, self._io)
+
+        argv = ["update"] + plugin_names
+        if self.option("dry-run"):
+            argv.append("--dry-run")
+
+        return update_command.run(
+            IO(
+                StringInput(" ".join(argv)),
+                self._io.output,
+                self._io.error_output,
+            )
+        )
+
+    def create_pyproject_from_package(
+        self, package: "ProjectPackage", path: "Path"
+    ) -> None:
+        if path.joinpath("pyproject.toml").exists():
+            return
+
+        import tomlkit
+
+        from poetry.layouts.layout import POETRY_DEFAULT
+
+        pyproject = tomlkit.loads(POETRY_DEFAULT)
+        content = pyproject["tool"]["poetry"]
+
+        content["name"] = package.name
+        content["version"] = package.version.text
+        content["description"] = package.description
+        content["authors"] = package.authors
+
+        dependency_section = content["dependencies"]
+        dependency_section["python"] = package.python_versions
+
+        for dep in package.requires:
+            constraint = tomlkit.inline_table()
+            if dep.is_vcs():
+                constraint[dep.vcs] = dep.source_url
+
+                if dep.reference:
+                    constraint["rev"] = dep.reference
+            elif dep.is_file() or dep.is_directory():
+                constraint["path"] = dep.source_url
+            else:
+                constraint["version"] = dep.pretty_constraint
+
+            if not dep.marker.is_any():
+                constraint["markers"] = str(dep.marker)
+
+            if dep.extras:
+                constraint["extras"] = list(sorted(dep.extras))
+
+            if len(constraint) == 1 and "version" in constraint:
+                constraint = constraint["version"]
+
+            dependency_section[dep.name] = constraint
+
+        path.joinpath("pyproject.toml").write_text(
+            pyproject.as_string(), encoding="utf-8"
+        )
+
+    def get_existing_packages_from_input(
```

Comment:
This might be better served as a reusable logic I feel

Reply:
Honestly, I don't think it warrants separating it from the command. It's a small function so I don't mind leaving it there.